content/resources/Networking/Colyseus_ammojs.md

```diff
@@ -0,0 +1,320 @@
+keywords: colyseus, multiplayer, networking, ammojs, server
+further-reading:
+video-overview:
+video-content:
+---
+![Wireframe](/img/resources/networking/colyseus_ammojs/d1.gif)
+
+This guide will help you realize the synchronization of network physical effects based on colyseus. This is a very simple demonstration. It does not include any server verification. Collision and detection only occur on the player's client. The server is responsible for synchronizing the data of physical effects and allocating the permissions of physical calculation;
+
+We use ammojs as the physical engine for this demonstration, and you can also replace it with other physical engines;
+
+
+## Full source code
+
+- [colyseus-babylonjs-ammojs](https://github.com/renjianfeng/colyseus-babylonjs-ammojs)
+
+## Before you start
+
+#### Prior Knowledge Expected
+
+-  ([Using A Physics Engine](https://doc.babylonjs.com/divingDeeper/physics/usingPhysicsEngine))
+-  ([Real-time Multiplayer with Colyseus](https://doc.babylonjs.com/guidedLearning/multiplayer/Colyseus))
+
+
+## Quick demo
+
+Please clone this project and follow the instructions to install and start the project:
+[colyseus-babylonjs-ammojs](https://github.com/renjianfeng/colyseus-babylonjs-ammojs)
+
+#### Client application
+
+To be able to build the client application, you'll need to enter in the folder,
+and install its dependencies first.
+
+```
+cd client/
+npm install
+```
+
+Now you can build and run it by running:
+
+```
+npm start
+```
+
+It will spawn the `webpack-dev-server`, listening on [http://localhost:8080](http://localhost:8080).
+
+
+#### Server application
+
+For the server, the steps are exactly the same. Install the dependencies:
+
+```
+cd server/
+npm install
+```
+
+Now you can build and run it by running:
+
+```
+npm start
+```
+
+It will spawn a web socket server, listening on [ws://localhost:2657](ws://localhost:2657).
+
+## Explanation of overall implementation logic
+The red mesh calculates the physical effects locally of the current user, and the green mesh calculates the physical effects on the clients of other players, and then synchronizes them through colyseus.
+
+the sphere represents the player character, and the cube represents the interactive objects in the scene.
+
+The first player to enter the scene is responsible for the physical calculation of the cube, and other players are responsible for receiving data and rendering. When other players collide with the cube, the physical calculation of the cube is transferred to the collided players. You can distinguish these changes by color, just like the difference between green and red mentioned above
+
+Yes, that's it
+
+## Code implementation demonstration
+
+#### Character control and network synchronization
+
+First, we create a box and ground, and add physics to it,the ground represents the scene, and the box represents the interactive objects in the scene (such as a football played by many people)
+
+```typescript
```

Comment:
typescript is not supported by markdown (AFAIK), this should be javascript for correctly parsing the markdown.

Reply:
ok